libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -69,7 +72,7 @@ internal class MapboxTripSession(
                 routeProgress = null
             }
             cancelOngoingUpdateNavigatorStatusDataJobs()
-            mainJobController.scope.launch {
+            setRouteJob = mainJobController.scope.launch {
                 navigator.setRoute(value)?.let {
                     routeAlerts = it.routeAlerts
                 }
```

Comment:
Could a fix be to move the trip session guard to be inside the coroutines? that way we're not blocking anything, only preventing the broadcast
``` kotlin
- if (state == TripSessionState.STARTED) {
     updateDataFromNavigatorStatus()
- }

private fun updateDataFromNavigatorStatus() {
  val updateNavigatorStatusDataJob = mainJobController.scope.launch {
    + if (state != TripSessionState.STARTED) return
    val status = getNavigatorStatus()
...
```

Reply:
But it seems the `cancelOngoingUpdateNavigatorStatusDataJobs` should also guard against this 🤔